File: app/src/main/java/com/arnold/uagroceries/ui/HostActivity.java
package com.arnold.uagroceries.ui;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.arnold.uagroceries.R;

public class HostActivity extends AppCompatActivity {

    private final String TAG = this.getClass().getSimpleName();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_host);


        switch (getIntent().getStringExtra("EXTRA")){
            case "Login":
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container_view, new LoginFragment()).commit();
                break;

            case "Register":
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container_view, new RegistrationFragment()).commit();
                break;


        }



    }

   @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }
}

File: app/src/main/java/com/arnold/uagroceries/ui/registration/HostActivity.kt
package com.arnold.uagroceries.ui.registration

import android.os.Bundle
import android.os.PersistableBundle
import androidx.appcompat.app.AppCompatActivity
import com.arnold.uagroceries.R

class HostActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        setContentView(R.layout.activity_home_activity)

        when (intent.getStringExtra("EXTRA")) {
            "Login" -> supportFragmentManager.beginTransaction().replace(R.id.fragment_container_view, LoginFragment()).commit()
            "Register" -> supportFragmentManager.beginTransaction().replace(R.id.fragment_container_view, RegistrationFragment()).commit()
        }


    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }
}